lib/http_accept_language/auto_locale.rb

```diff
@@ -11,7 +11,9 @@ module AutoLocale
     private
 
     def set_locale
-      I18n.locale = http_accept_language.compatible_language_from(I18n.available_locales)
+      hal = http_accept_language
+      hal.header ||= I18n.default_locale.to_s
+      I18n.locale = hal.compatible_language_from(I18n.available_locales)
```

Comment:
OK, it's your call to decide how HttpAcceptLanguage::AutoLocale behaves.  You could easily implement your own variant to meet your needs anyway, like refusing a fallback by returning 406, adding a Content-Language response header, etc.